ui/job-view/details/BugFiler.jsx

```diff
@@ -264,25 +264,27 @@ export class BugFilerClass extends React.Component {
       if (lowerJobGroupName.includes('web platform')) {
         failurePath = failurePath.startsWith('mozilla/tests')
           ? `testing/web-platform/${failurePath}`
-          : `testing/web-platform/tests/${failurePath}`;
+          : `testing/web-platform/tests${failurePath}`;
       }
 
       // Search mercurial's moz.build metadata to find products/components
       fetch(
         `${hgBaseUrl}mozilla-central/json-mozbuildinfo?p=${failurePath}`,
       ).then(resp =>
         resp.json().then(firstRequest => {
+          if (firstRequest.error) {
+            notify("Unable to infer product/component via metadata. Please manually search for a product.");
+          }
           if (
-            firstRequest.data.aggregate &&
+            firstRequest.data && firstRequest.data.aggregate &&
             firstRequest.data.aggregate.recommended_bug_component
           ) {
             const suggested =
               firstRequest.data.aggregate.recommended_bug_component;
             suggestedProductsSet.add(`${suggested[0]} :: ${suggested[1]}`);
           }
-
           // Make an attempt to find the file path via a dxr file search
-          if (suggestedProductsSet.size === 0 && possibleFilename.length > 4) {
+          if (suggestedProductsSet.size === 0 && possibleFilename && possibleFilename.length > 4) {
```

Comment:
possibleFilename was sometimes undefined because it was only ever assigned in the if() block checking if the failure started with "TEST-UNEXPECTED". That should be fixed by changes above, but might as well make sure it's defined at this point.